src/app/input-bar/input-bar.component.ts

```diff
@@ -18,25 +19,165 @@ export class InputBarComponent implements OnInit {
 
   ngOnInit(): void { }
 
-  pushInput() {
+  pushInput(): void {
     const options = this.parseInput(this.newInput);
-    this.ccdService.updateRegex(options.regex);
-    this.ccdService.updateGroupingKey(options.group);
+    this.ccdService.updateFilters(options.filters);
+    this.ccdService.updateGroupingKey(options.groupBy);
 
-    // we delete any options that are not filter by metadata keys so that
-    // only filter keys are left for the filter pipe to manage.
-    delete options.regex;
-    delete options.group;
-    this.ccdService.updateFilters(options);
+    // We need to scan filters for regex so we can highlight matches
+    options.filters.forEach((filter: FilterOption<any>) => {
+      if (filter.filterKey === FilterKeys.Regex && !filter.negation) {
+        const regex = new RegExp(`(${filter.inputString})`, 'g');
+        this.ccdService.updateRegex(regex);
+      }
+    });
 
   }
 
   parseInput(input: string) {
```

Comment:
Is it easy to create a unit test especially for this function?

Reply:
Yeah, before we run the function we just need to set newInput and subscribe to the various observables

Then we can run the function and confirm that the subscribers receive the filters and groupings that we expect